tick/survival/convolutional_sccs.py

```diff
@@ -0,0 +1,922 @@
+        else:
+            # Default prox: does nothing
+            proxs = [ProxZero()]
+
+        prox_obj = ProxMulti(tuple(proxs))
+
+        return prox_obj
+
+    def _detect_support(self, coeffs):
+        """Return the ranges over which consecutive coefficients are equal in
+        case at least two coefficients are equal.
+
+        This method is used to compute the ranges for ProxEquality,
+        to enforce a support corresponding to the support of `coeffs`.
+
+         example:
+         coeffs = np.array([ 1.  2.  2.  1.  1.])
+         self._detect_support(coeffs)
+         >>> [(1, 3), (3, 5)]
+         """
+        kernel = np.array([1, -1])
+        groups = []
+        for i in self.penalized_features:
+            idx = int(self._features_offset[i])
+            n_lags = int(self._n_lags[i])
+            if n_lags > 0:
+                acc = 1
+                for change in np.convolve(coeffs[idx:idx+n_lags+1], kernel,
+                                          'valid'):
+                    if change:
+                        if acc > 1:
+                            groups.append((idx, idx + acc))
+                        idx += acc
+                        acc = 1
+                    else:
+                        acc += 1
+                # Last coeff always count as a change
+                if acc > 1:
+                    groups.append((idx, idx + acc))
+        return groups
+
+    @staticmethod
+    def _construct_solver_obj(step, max_iter, tol, print_every,
+                              record_every, verbose, seed):
+        # seed cannot be None in SVRG
+        solver_obj = SVRG(step=step, max_iter=max_iter, tol=tol,
+                          print_every=print_every,
+                          record_every=record_every,
+                          verbose=verbose, seed=seed)
+
+        return solver_obj
+
+    def _construct_generator_obj(self, strength_tv_range,
+                                 strength_group_l1_range,
+                                 logspace=True):
+        generators = []
+        if len(strength_tv_range) == 2:
+            if logspace:
+                generators.append(Log10UniformGenerator(*strength_tv_range))
+            else:
+                generators.append(uniform(strength_tv_range))
+        else:
+            generators.append(null_generator)
+
+        if len(strength_group_l1_range) == 2:
+            if logspace:
+                generators.append(
+                    Log10UniformGenerator(*strength_group_l1_range))
+            else:
+                generators.append(uniform(strength_group_l1_range))
+        else:
+            generators.append(null_generator)
+
+        return generators
+
+    # Properties #
+    @property
+    def step(self):
+        return self._step_size
+
+    @step.setter
+    def step(self, value):
+        self._set('_step_size', value)
+        self._solver_obj.step = value
+
+    @property
+    def _strengths(self):
```

Comment:
I'd rather like a separate C_tv and C_group_l1 with getters and setter, where we check that there are > 0, and where we update private _strenght_tv and _strengh_group_l1 (if zero we put None). This is what we do in linear learners

Reply:
this property is meant to be private, i.e. used only internally to avoid having many .set() everywhere, C_tv and C_group_l1 have public getters and setters of their own. I can remove it if you find it ugly